gemrb/core/Spellbook.cpp

```diff
@@ -395,13 +395,13 @@ unsigned int Spellbook::GetKnownSpellsCount(int type, unsigned int level) const
 //this one purges all instances of known spells of the same name from memory
 void Spellbook::RemoveMemorization(CRESpellMemorization* sm, const ResRef& resRef)
 {
-	for (auto ms = sm->memorized_spells.begin(); ms != sm->memorized_spells.end(); ++ms) {
+	for (auto ms = sm->memorized_spells.begin(); ms != sm->memorized_spells.end(); ) {
 		if (resRef != (*ms)->SpellResRef) {
+			ms++;
```

Comment:
prefix operators are preferred, since they're a tiny bit more efficient, which is why this would generate a "code smell" warning on Sonar.